include/benchmark/benchmark.h

```diff
@@ -241,6 +241,32 @@ BENCHMARK(BM_test)->Unit(benchmark::kMillisecond);
 #define BENCHMARK_GCC_VERSION (__GNUC__ * 100 + __GNUC_MINOR__)
 #endif
 
+#if defined(__clang__)
+  #if !defined(COMPILER_CLANG)
+    #define COMPILER_CLANG
+  #endif
+#elif defined(_MSC_VER)
+  #if !defined(COMPILER_MSVC)
+    #define COMPILER_MSVC
+  #endif
+#elif defined(__GNUC__)
+  #if !defined(COMPILER_GCC)
+    #define COMPILER_GCC
+  #endif
+#endif
+
+#ifndef __has_builtin
+#define __has_builtin(x) 0
+#endif
+
+#if defined(COMPILER_GCC) || __has_builtin(__builtin_unreachable)
```

Comment:
given the rest of this header (above) works on __clang__ and __GNUC__ directly, perhaps you don't need to define the COMPILER_XXX macros?

If you do want to then maybe define them higher up (line 188?) and use them throughout.

But i think it's fine to remove lines 244 to 256 and not define new macros just for this check.